### PR TITLE
Revert removal of `__conda_reactivate`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Reverts https://github.com/conda/conda/pull/14277

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/shell/etc/profile.d/conda.sh [hunk 1]
```diff
@@ -27,6 +27,15 @@ __conda_activate() {
     __conda_hashr
 }
 
+__conda_reactivate() {
+    # FUTURE: conda 25.9, remove this function
+    echo `__conda_reactivate` is deprecated and will be removed in 25.9. Use `__conda_activate reactivate` instead. 1>&2
```
[the remark this answers is not in the document]
Comment (dholth) [reply conda/shell/etc/profile.d/conda.sh:32]: Can we keep __conda_reactivate and have it call __conda_activate reactivate?

[the remark this answers is not in the document]
Comment (kenodegard) [reply conda/shell/etc/profile.d/conda.sh:32]: I don't think we should keep redundant code around, but if others think we should keep it I can remove the deprecation

[the remark this answers is not in the document]
Comment (dholth) [reply conda/shell/etc/profile.d/conda.sh:32]: To me aliases for compatibility are pretty good, and not difficult to maintain.

[the remark this answers is not in the document]
Comment (jjerphan) [reply conda/shell/etc/profile.d/conda.sh:32]: Unfortunately, this causes problems for users with recursive calls… (see https://github.com/conda-forge/miniforge/issues/700#issuecomment-2568526971).

This must have been:
```suggestion
    # FUTURE: conda 25.9, remove this function
    echo "\'__conda_reactivate\' is deprecated and will be removed in 25.9. Use \'__conda_activate reactivate\' instead." 1>&2
```

I do not think that removing aliases is a priority at all.

[the remark this answers is not in the document]
Comment (jaimergp) [reply conda/shell/etc/profile.d/conda.sh:32]: Thanks @jjerphan, sending a fix.

[the remark this answers is not in the document]
Comment (jaimergp) [reply conda/shell/etc/profile.d/conda.sh:32]: https://github.com/conda/conda/pull/14468

**Pull request merged**